In [1]:
import pandas as pd
import math

# **MODEL CREATION** #

''' Hacer orimero que cada palabra de la columna messages sea una linea de un nuevo dataframe.
mirar las palabras mas repetidas.
Asociar valores a las palabras y crear dataframe
en mi dataframe pero copia poner columna por cada tag que encuentre.
despues a partir de aquí asociarle emociones
y construir el modelo.'''

'''NLP LEMMATIZATION
generar nova columna per lemmatization stemming'''

'''Trobar formula del sample minim que necessito per ser representatiu
per agafar el sample random 
llistat emocions dataframe

dels 1000 missatges anar un per un i classificar

entreno sobre els 1000 per trobar el millor model

entreno despres sobre tot el sample i despres ja li poso a tot el meu dataset

fer bootstrapping i fer el m, bootstrapping statistics'''

x boost, random forest, 
amarquescancio@gmail.com

In [2]:
df = pd.read_csv("../data/translation/completo.csv")

In [3]:
df.shape

(118752, 4)

In [4]:
import math

# Datos
N = 118752  # Tamaño total de la población
Z = 1.96    # Valor crítico para un nivel de confianza del 95%
E = 0.05    # Margen de error del 5%
p = 0.5     # Estimación de la proporción

# Fórmula para calcular el tamaño de la muestra
n = (N * Z**2 * p * (1 - p)) / ((E**2 * (N - 1)) + Z**2 * p * (1 - p))

# Redondear hacia arriba ya que no puedes tener fracciones de personas en tu muestra
n = math.ceil(n)

print(f"Min necessary sample: {n}")


Min necessary sample: 383


Even though it says the minimmum sample can be 383 I'm going to catch 1000 to be sure to cover all 3 languages.

audio, image, video = neutral

In [27]:
def get_random_df(df):

    df = df[(df["message"].str.lower() != "\u200eaudio ") & (df["message"].str.lower() != "\u200evideo ")]
    df = df[df["message"].str.lower() != "\u200eimage "] 

    sample_df = df.sample(n=1000, random_state=42, ignore_index=True)
    return sample_df

sdf = get_random_df(df)

In [18]:
sdf.iloc

In [28]:
sdf.sample(30)

date      time name  \
41   2018-12-13  12:53:57   Em   
771  2021-07-22  12:00:55   Em   
828  2021-07-09  13:53:54   Em   
152  2017-10-10  18:16:19   Em   
407  2023-01-08  19:17:40   Em   
964  2017-06-16  20:27:10   Em   
940  2019-12-25  20:22:54   Em   
529  2020-12-16  11:53:58   Em   
883  2021-12-10  23:36:21   Em   
64   2021-06-06  12:43:18   Em   
143  2019-12-14  11:57:18   Em   
39   2023-07-17  14:55:39   Em   
461  2021-05-04  12:29:51   Em   
929  2017-05-07  13:49:47   Em   
86   2018-02-10  00:12:15   Em   
864  2021-02-07  23:22:08   Em   
556  2017-12-10  12:46:52   Em   
219  2018-10-07  16:24:32   Em   
518  2021-06-07  23:38:47   Em   
710  2023-05-07  18:28:31   Em   
870  2021-03-07  20:42:38   Em   
507  2019-10-05  19:58:43   Em   
377  2019-05-23  08:48:35   Em   
102  2017-06-22  13:57:17   Em   
40   2020-10-19  09:19:33   Em   
116  2017-07-31  02:22:48   Em   
946  2020-09-30  15:38:33   Em   
923  2021-06-09  10:18:16   Em   
227  2019-02-19  20:48:58   Em   
815  2020-11-02  12:18:36   Em   

                                               message  
41                                           Hola amor  
771                         ‎You deleted this message.  
828                                            Si o no  
152                                       Sisi tranqui  
407             Amigaaa perooo quieres que nos veamos?  
964                               Dormimos juntos hoy?  
940                                    A veure que tal  
529                                             Thxxxx  
883                                          ‎sticker   
64                                   Seria en espanyol  
143                  Donde quedamos para encontrarnos?  
39                                   David Lop Serrano  
461                                                 Ja  
929                                   Te echa de menos  
86   Tiaaaa hem vingut a veureus a la portaa del cinee  
864  Pero com no tinguis camaras a casa meva dubto ...  
556                                                Tu?  
219                                  No te estreseeees  
518  Ja esta para de mirar i tanca laplicacio i ja ...  
710                              Nono que esta dormint  
870         Jo crec que t’anira molt be aquest monoleg  
507                                            Ueeeeee  
377                        I em podras portar berenar?  
102       Porque no venias conmigo a casa mis abuelos?  
40                                          Que macooo  
116                         Heeeu que haces despierto?  
946                  Ajajjaajajaj sissiiii totalmeeent  
923                                    Donde estaaaas?  
227                                 Que ha pasado tia?  
815                 Tia que me estoy haciendo oklas ya

In [25]:
sdf['classification'] = None

In [31]:
sdf.to_csv("../data/translation/to_classify.csv")

In [39]:
emotions= [{"sadness":1},
           {"joy":2},
           {"fear":3},
           {"anger":4},
           {"surprise":5},
           {"defensive":6},
           {"nostalgia":7},
           {"hapiness":8},
           {"stressed":9},
           {"expectant":10},
           {"insecure":11},
           {"assertivity":12},
           {"neutral":13},
           {"excitement":14},
           {"comprehensive":15},
           {"helpless":16},
           {"love":17},
           {"deception":18},
           {"doubtful":19},
           {"picarona":20}]

In [ ]:
11